cli_tools/common/image/importer/bootable_disk_processor.go

```diff
@@ -40,10 +42,7 @@ func (b *bootableDiskProcessor) process(pd persistentDisk) (persistentDisk, erro
 	if err != nil {
 		b.logger.User("Finished making disk bootable")
 		daisy_utils.PostProcessDErrorForNetworkFlag("image import", err, b.request.Network, b.workflow)
-		err = customizeErrorToDetectionResults(b.request.OS,
-			b.workflow.GetSerialConsoleOutputValue("detected_distro"),
-			b.workflow.GetSerialConsoleOutputValue("detected_major_version"),
-			b.workflow.GetSerialConsoleOutputValue("detected_minor_version"), err)
+		err = customizeErrorToDetectionResults(b.request.OS, b.detectedOs, err)
```

Comment:
Since this code relies on receiving an error from the processor, does this mean we have to have OS detection in two places: the new inspector as well as in translate? If so, that seems fragile and could potentially lead to misleading results (i.e. translate failed for other reasons but we print an OS mismatch error based on the inspection results in the previous step). 

Can't we fail early, even before translation starts, since we already know about OS mismatch at that point?

Reply:
The net change of this PR is to reduce two OS detections to one. The results of the first OS detection are *used* by the processor, rather than the processor performing its own detection.

Regarding fragility, if `plan.detectedOs` is nil, then the cusotmized error message won't be generated and the user won't will be shown the original error message.

All of that said, I'm curious about this idea: "Can't we fail early, even before translation starts, since we already know about OS mismatch at that point?". Originally I had envisioned the `--os` flag as an escape hatch, that allowed the user to override the value from OS detection. Without this hatch, I'm worried about creating a situation where the user is stuck and unable to import their image (assuming there's a problem with OS detection).

Reply:
My point about having two detections is because `err` is coming from translation, while detectedOs is coming from inspection. This code relies on inspection returning an error, assuming this error is for a mismatched OS, means we detect OS twice. Can we avoid this and just use detected OS and --os flag value before translation even starts?

Reply:
Talked offline. 

I checked, and the original error message is shown for image import, but not OVF import (since OVF import has stricter filtering in the gcloud). So I updated `customizeErrorToDetectionResults` to always emit the original error at the user level, to ensure that it's shown in OVF output.